### PR TITLE
WB-1569: Implement useScopedTheme hook

### PR DESCRIPTION
**Pull request by jandrade**
## Summary:

This PR implements the `useScopedTheme` hook, which will allow a component to
get and use the theme internally. This is useful for components that need to
render a themeable component, but don't want to pass the theme down to the
component. For example, a `Button` component might render a `LabelMedium`
component, but the `LabelMedium` component should not be themed by the
`Button`'s theme.

### Implementation plan:

1. #1975
2. #1977
3. **WB-1569: Implement useScopedTheme hook [CURRENT]**
4. #1979 
5. #1981
6. #1982
7. Create Storybook example for adding theming support to a custom component (WIP)

Issue: WB-1569

## Test plan:

Navigate to /?path=/docs/theming-usescopedtheme--docs and verify that the
`useScopedTheme` hook is working as expected and that the documentation is
clear.

https://5e1bf4b385e3fb0020b7073c-podatcwryl.chromatic.com/?path=/docs/theming-usescopedtheme--docs

Screenshot 2023-08-10 at 6 04 00 PM

### CODE REVIEW

File: __docs__/wonder-blocks-theming/use-scoped-theme.mdx [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+import {tokens, useScopedTheme} from "@khanacademy/wonder-blocks-theming";
+
+import {ViewWithTheme} from "./theme-examples";
+
+<Meta title="Theming / useScopedTheme" />
+
+# `useScopedTheme`
+
+`useScopedTheme` is a custom hook that allow us to get the theme based on a
```
Comment (nishasy) [inline __docs__/wonder-blocks-theming/use-scoped-theme.mdx:11]: super nit
```suggestion
`useScopedTheme` is a custom hook that allows us to get the theme based on a
```

File: __docs__/wonder-blocks-theming/use-scoped-theme.mdx [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+  },
+  ...,
+};
+
+const themes = {
+  default: defaultTheme,
+};
+
+// 👉 2. Create the theme context scoped to this component.
+const ComponentThemeContext = React.createContext(themes.default);
```
Comment (nishasy) [inline __docs__/wonder-blocks-theming/use-scoped-theme.mdx:41]: Are you planning to change this to use the custom `createThemeContext` that you wrote in #1977?

**Review by nishasy — COMMENTED** [review]
Looks good to me! It could probably use another pair of eyes on it though, since I haven't personally written a hook myself before.

**Comment by jandrade** [comment]
The parent pull-request ([#1977](https://github.com/Khan/wonder-blocks/pull/1977)) has been merged into `main`, but this branch (`theming-scoped`) now has conflicts with the new base branch. These conflicts must be resolved before checks can complete on this pull-request.

File: __docs__/wonder-blocks-theming/use-scoped-theme.mdx (hunk 2 above)
Comment (jandrade) [reply __docs__/wonder-blocks-theming/use-scoped-theme.mdx:41]: ohh good point! I'll replace it here.

**Pull request merged**